float64 battery_percentage
string is_charging